twitch4j/src/main/java/com/github/twitch4j/TwitchClientPoolBuilder.java

```diff
@@ -0,0 +1,447 @@
+package com.github.twitch4j;
+
+import com.github.philippheuer.credentialmanager.CredentialManager;
+import com.github.philippheuer.credentialmanager.CredentialManagerBuilder;
+import com.github.philippheuer.credentialmanager.domain.OAuth2Credential;
+import com.github.philippheuer.events4j.api.service.IEventHandler;
+import com.github.philippheuer.events4j.core.EventManager;
+import com.github.philippheuer.events4j.simple.SimpleEventHandler;
+import com.github.twitch4j.auth.TwitchAuth;
+import com.github.twitch4j.chat.ITwitchChat;
+import com.github.twitch4j.chat.TwitchChat;
+import com.github.twitch4j.chat.TwitchChatBuilder;
+import com.github.twitch4j.chat.TwitchChatConnectionPool;
+import com.github.twitch4j.common.config.ProxyConfig;
+import com.github.twitch4j.common.config.Twitch4JGlobal;
+import com.github.twitch4j.common.util.EventManagerUtils;
+import com.github.twitch4j.common.util.ThreadUtils;
+import com.github.twitch4j.extensions.TwitchExtensions;
+import com.github.twitch4j.extensions.TwitchExtensionsBuilder;
+import com.github.twitch4j.graphql.TwitchGraphQL;
+import com.github.twitch4j.graphql.TwitchGraphQLBuilder;
+import com.github.twitch4j.helix.TwitchHelix;
+import com.github.twitch4j.helix.TwitchHelixBuilder;
+import com.github.twitch4j.kraken.TwitchKraken;
+import com.github.twitch4j.kraken.TwitchKrakenBuilder;
+import com.github.twitch4j.pubsub.ITwitchPubSub;
+import com.github.twitch4j.pubsub.TwitchPubSub;
+import com.github.twitch4j.pubsub.TwitchPubSubBuilder;
+import com.github.twitch4j.pubsub.TwitchPubSubConnectionPool;
+import com.github.twitch4j.tmi.TwitchMessagingInterface;
+import com.github.twitch4j.tmi.TwitchMessagingInterfaceBuilder;
+import feign.Logger;
+import io.github.bucket4j.Bandwidth;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.With;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+/**
+ * Builder to get a TwitchClientPool Instance by provided various options, to provide the user with a lot of customizable options.
+ */
+@Slf4j
+@Getter
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class TwitchClientPoolBuilder {
```

Comment:
Should this share a common interface or abstract class with TwitchClientBuilder?

Reply:
We don't need that here, that class is only a workaround because we don't want to change the return types in the TwitchClientBuilder, so this class will be removed with the next major release and those 4 new properties will go into the TwitchClientBuilder.